### ISSUE TITLE
gdal can not build with odbc on window

### ISSUE THREAD

**Issue by liyq0307** [conan-io/conan-center-index#26210]
### What is your question?

gdal can not build with odbc on window,What should I do?

**Comment by liyq0307** [comment]
> Hi @liyq0307 - which conan command are you running, on which platform/versions (see the issue template), and what is the output? Thanks!

When I compile the gdal library with the command "conan create . gdal/3.8.3@ -s compiler="Visual Studio" -s build_type=Release -s arch=x86_64 -s compiler.version=17", if "with_odbc": True is set, the error : "here are invalid packages (packages that cannot exist for this configuration):odbc/2.3.11: Invalid ID: odbc is a system lib on Windows will be thrown"; if "with_odbc": True is set and self.requires("odbc/2.3.11") is commented out:" CMake Error at src/cmake/helpers/CheckDependentLibraries.cmake:201 (message):Configured to use ODBC", but not found, what should I do in Windows?

**Comment by SpaceIm** [comment]
Indeed, it's an issue of gdal >= 3.5.0 recipe, it doesn't properly handle odbc on Windows. When host OS is Windows, it doesn't make sense to depend on odbc recipe (which is actually unixodbc), odbc library is a system library on Windows. It can be fixed by using the same logic in gdal < 3.5.0 recipe.

**Comment by liyq0307** [comment]
> Indeed, it's an issue of gdal >= 3.5.0 recipe, it doesn't properly handle odbc on Windows. When host OS is Windows, it doesn't make sense to depend on odbc recipe (which is actually unixodbc), odbc library is a system library on Windows.
> 
> It can be fixed by using the same logic as gdal < 3.5.0 recipe:
> 
> https://github.com/conan-io/conan-center-index/blob/4622ac85d1cec8cb7a2fcc8a1796d4b73bff285e/recipes/gdal/pre_3.5.0/conanfile.py#L276-L277
> 
> 
> https://github.com/conan-io/conan-center-index/blob/4622ac85d1cec8cb7a2fcc8a1796d4b73bff285e/recipes/gdal/pre_3.5.0/conanfile.py#L836-L843

Thanks for your reply！
I followed your method and used gdal < 3.5.0 to fix the script, but when setting with_odbc=True, the following error is still thrown:
-- Found ODBC: 0
--   ODBC_TARGET:
--   ODBC_LIBRARIES:
--   ODBC_INCLUDE_DIRS:
CMake Error at src/cmake/helpers/CheckDependentLibraries.cmake:201 (message):
  Configured to use ODBC, but not found
Call Stack (most recent call first):
  src/cmake/helpers/CheckDependentLibraries.cmake:274 (gdal_check_package)
  src/gdal.cmake:265 (include)
  src/CMakeLists.txt:264 (include)

**Comment by SpaceIm** [comment]
I think patches in this recipe prevent to find system odbc on windows (injection of this cmake file specifically https://github.com/conan-io/conan-center-index/blob/master/recipes/gdal/post_3.5.0/cmake/ConanFindPackage.cmake).
I don't have the time to dig into this, you can try to ping maintainers of gdal >= 3.5.0 recipe.
